packages/fluentui/react-builder/src/components/ReaderNarration.tsx

```diff
@@ -16,98 +17,119 @@ export type ReaderNarrationProps = {
+  const handleElementPathChange = (event: any, props: DropdownProps) => {
+    selectedElementPath = props.value as string;
+    prevNarrationElement = narrationElement;
+    setNarrationElement(focusableElements[selectedElementPath]);
+  }; // End handleElementPathChange
+
+  // Handles the "focusin" event by updating the current and previous narration elements.
   const handleFocusIn = React.useCallback(
     event => {
-      computeAndSave(event.target as IAriaElement);
+      prevNarrationElement = narrationElement;
+      setNarrationElement(event.target as IAriaElement);
     },
-    [computeAndSave],
+    [narrationElement],
   ); // End handleFocusIn
 
-  // Handles the narration path dropdown change event by saving the narration path.
-  const handleNarrationPathChange = (event: any, props: DropdownProps) => {
-    narrationPath = props.value as string;
-    setSelectedNarrationPath(narrationPath);
-    const text = narrationTexts[narrationPath];
-    setCompleteText(text);
-  }; // End handleNarrationPathChange
-
-  // Recomputes the narration paths and texts upon every render.
+  // Recomputes the narration text upon every render.
   React.useEffect(() => {
-    if (inUseMode || !ref.current || aomMissing) {
+    if (!ref.current || aomMissing) {
       return;
     }
 
-    // Compute and save the narration paths and texts for the selected component's parent element and its focusable descendants
-    const element = ref.current.ownerDocument.querySelector(selector) as IAriaElement;
-    computeAndSave(element);
-  }); // End useEffect
+    // Recompute and save the focusable elements and their paths for the tree rooted at the selector's element upon every selector change
+    if (prevSelector !== selector) {
+      // Begin if 1
+      const element = ref.current.ownerDocument.querySelector(selector) as IAriaElement;
+      computer.getFocusableElements(element).then(focusableElementsItems => {
+        focusableElements = {};
+        elementsPaths = [];
+        focusableElementsItems.forEach(focusableElementItem => {
+          // Begin forEach 1
+          const path = focusableElementItem.path.join(' > ');
+          focusableElements[path] = focusableElementItem.element;
+          elementsPaths.push(path);
+        }); // End forEach 1
+        prevNarrationElement = narrationElement;
+        if (elementsPaths.length >= 1) {
+          // Begin if 2
+          // Preselect the path and the narration element to the first focusable element with tabindex >= 0
+          let preselectedElementItem = focusableElementsItems[0];
+          focusableElementsItems.some(focusableElementItem => {
+            // Begin some 1
+            if (focusableElementItem.element.tabIndex >= 0) {
+              // Begin if 3
+              preselectedElementItem = focusableElementItem;
+              return true;
+            } // End if 3
+            return false;
+          }); // End some 1
```

Comment:
You generally shouldn't be using some and every for side effects. Especially when the boolean they return has no meaning in the code

Do you need to support IE11 ? this might be simpler

```suggestion
         
          let preselectedElementItem = focusableElementItems.find(item => item.element.tabIndex >=0) ||  focusableElementsItems[0];
```

Reply:
Thanks, that's a good one.